python/src/aiconfig/editor/client/src/components/JSONEditorToggleButton.tsx

```diff
@@ -10,8 +12,11 @@ export default function JSONEditorToggleButton({
   isRawJSON,
   setIsRawJSON,
 }: Props) {
+  const { readOnly } = useContext(AIConfigContext);
+
+  const toggleJSONButtonLabel = !readOnly ? "Toggle JSON editor" : "View JSON";
```

Comment:
nit, prefer flipping so no negation is needed
```
const toggleJSONButtonLabel = readonly ? "View JSON" : "Toggle JSON Editor";

Reply:
For context Ankush, I did it the other way for things like `!readOnly ? <addPromptButton>... : undefined` because I think it's clearer to show if the last value is undefined or null, but for this case I agree with Ryan

Reply:
updated